duet/impl.py

```diff
@@ -376,13 +378,22 @@ def tick(self):
             self._interrupted = False
 
         deadline = self.get_next_deadline()
-        timeout = None if deadline is None else max(deadline - self.time(), 0)
-        try:
-            ready_tasks = self._ready_tasks.get_all(timeout)
-        except TimeoutError:
-            for task in self.get_deadline_tasks(deadline):
-                task.interrupt(task, TimeoutError())
+        if deadline is None:
             ready_tasks = self._ready_tasks.get_all(None)
+        else:
+            ready_tasks: List[Task] = []
+            for i in itertools.count():
+                timeout = deadline - self.time()
```

Comment:
Sorry for the obtuse question, but is there any guarantee that `deadline - self.time() < threading.TIMEOUT_MAX`?

Reply:
No guarantee here, but the timeout is clipped inside `get_all` before being passed to `Condition.wait`, which is where problems can happen.